File: example-app/src/main/java/com/nickmcdowall/example/repository/FishRepositoryJpa.java
package com.nickmcdowall.example.repository;

import com.nickmcdowall.example.entity.Fish;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;

@Transactional
@Repository
public interface FishRepositoryJpa extends CrudRepository<Fish, String> {
    Fish findFishByName(String name);

    void deleteByName(String name);
}

File: example-app/src/main/java/io/lsdconsulting/intercceptors/example/repository/FishRepositoryJpa.kt
package io.lsdconsulting.intercceptors.example.repository

import io.lsdconsulting.intercceptors.example.entity.Fish
import jakarta.transaction.Transactional
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Transactional
@Repository
interface FishRepositoryJpa : CrudRepository<Fish?, String?> {
    fun findFishByName(name: String?): Fish?
    fun deleteByName(name: String?)
}
